### DISCUSSION TITLE
Vorschlag: Konfiguration Eingabedialog für Medium / Suggestion: Configuration input dialogue for medium

### DISCUSSION THREAD

**Discussion by LeAchim59** [Larian77/Openbiblio#31] [Ideas]
In unserer Ludothek wird immer wieder der Wunsch geäußert, doch die Liste der Eingabefelder bei einem neuen Medium zu beschränken, da im Besonderen die vielen eher für Bücher relevanten Felder nicht erfasst werden.
Definiert ist diese Liste in catalog/biblio_fields.php.
Natürlich ist es möglich, die nicht gewünschten Felder in dieser Datei einfach zu löschen. Das muss jeder aber dann nach einem Update wieder machen. (Bibliotheken interessieren die Werte für Spiele wahrscheinlich auch nicht.)

Es gäbe zwei Möglichkeiten, das Problem zu lösen: Entweder man arbeitet (zB wie Wordpress) mit alternativen Templates, d.h. man holt sich die Zeilen 129-173 aus einer anderen Datei, die nicht Teil des Grundsystems ist und individuell anpassbar ist (und bei Updates vielleicht nur geringfügig vom jeweiligen Admin angepasst werden muss) oder

Wir führen eine Konfigurationstabelle ein, in welcher die nicht konstanten Parameter aus den Aufrufen von printUsmarcInputText() eingetragen werden können, so dass sich jede Installation die Eingabemaske individuell zusammenstellen kann.
Ich bitte um Kommentare, was Ihr bevorzugen würde.,

---
In our games library, the wish is repeatedly expressed to limit the list of input fields for a new medium, as the many fields that are more relevant for books in particular are not recorded.
This list is defined in catalog/biblio_fields.php.
Of course, it is possible to simply delete the unwanted fields in this file. However, everyone will have to do this again after an update. (Libraries are probably not interested in the values for games either).

There are two ways to solve the problem: Either you work (e.g. like Wordpress) with alternative templates, i.e. you get the lines 129-173 from another file that is not part of the basic system and can be customised individually (and perhaps only needs to be adjusted slightly by the respective admin during updates) or

We introduce a configuration table in which the non-constant parameters from the calls to printUsmarcInputText() can be entered so that each installation can compile the input mask individually.
Please let me know what you would prefer.

Translated with DeepL.com (free version)

**Comment by superudo** [comment]
Ich könnte mir vorstellen, in der Bearbeitung der MARC-Felder für die Medientypen außer der Verwaltung nutzerdefinierter Felder alle standardmäßig definierten Felder anzuzeigen und jedem Feld eine Checkbox hinzufügt, mit der ausgewählt werden kann, ob dieses Feld in der Eingabemaske zu sehen ist oder nicht. 

---

I could imagine showing all standardly defined fields for the media types in the editing of MARC fields, in addition to managing custom fields, and adding a checkbox to each field with which you can select whether this field should be visible in the input mask or not.

**Comment by LeAchim59** [comment]
Noch variabler wäre eine Methode, die ich beim CRM-System Sugar (auch php) kennengelernt habe. Dort kann man für die Darstellung eines Datensatzes die darzustellenden Felder in eine Liste eintragen, das entscheidet nicht nur über das Anzeigen eines Feldes sondern auch über die Reihenfolge der Felder. Implementieren kann man das mit zwei Listen, die nebeneinander dargestellt werden. Die linke enthält zu Beginn alle möglichen Felder. Dann bewegt man einzelne Felder in die rechte, welche die darzustellenden Felder enthält. In dieser kann man außerdem Felder nach oben bzw. nach unten bewegen oder löschen.

--
An even more flexible approach is one I encountered in the Sugar CRM system (also built with PHP). There, to display a record, you can enter the fields to be displayed in a list; this determines not only which fields are shown but also their order. This can be implemented using two lists displayed side by side. The left pane initially contains all possible fields. You then move individual fields to the right pane, which contains the fields to be displayed. In this pane, you can also move fields up or down or delete them.

**Reply by superudo** [reply]
Das ist eine sehr gute Idee! Da wäre ich sofort dabei. Den Selector kann man ganz gut per Javascript implementieren, und ob wir in der Datenbank einen Wahrheitswert oder eine Zahl für die Reihenfolge abspeichern, macht keinen Unterschied.

---

That's a great idea! I'd be all for it. The selector can be implemented quite easily using JavaScript, and it doesn't matter whether we store a boolean value or a number in the database to determine the order.
